pontoon/base/models.py

```diff
@@ -1627,6 +1631,21 @@ def with_status_counts(self, locale):
                     default=0
                 )
             ),
+            rejected_count=Sum(
+                Case(
+                    When(
+                        Q(
+                            translation__locale=locale,
+                            translation__approved=False,
+                            translation__fuzzy=False,
+                            translation__rejected=True,
```

Comment:
I didn't notice any impact on performance. Since we don't have any DB constraints, I'd rather stay on the safe side and keep all the checks.